src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -395,7 +395,10 @@ public IPrincipal GetThreadPrincipal()
                         if (s_getUnauthenticatedPrincipal == null)
                         {
                             Type type = Type.GetType("System.Security.Principal.GenericPrincipal, System.Security.Claims", throwOnError: true);
-                            Volatile.Write(ref s_getUnauthenticatedPrincipal, (Func<IPrincipal>)Delegate.CreateDelegate(typeof(Func<IPrincipal>), type, "GetDefaultInstance"));
+                            // Don't throw PNSE if null like for WindowsPrincipal as UnauthenticatedPrincipal should
+                            // be available on all platforms.
+                            Volatile.Write(ref s_getUnauthenticatedPrincipal,
```

Comment:
You should also delete `ignoreCase: false, throwOnBindFailure: false` and keep it as it used to be so that it throws exception when things go wrong.